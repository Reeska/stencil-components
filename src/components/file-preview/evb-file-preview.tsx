import { Component, Prop } from "@stencil/core";

@Component({
  tag: 'evb-filepreview',
  styleUrl: 'evb-filepreview.css',
  shadow: true
})
export class EvbFilePreview {

  @Prop() src: string;
  @Prop() alt: string = 'preview';
  @Prop() caption?: string;

  render() {
    if (this.src) {
      const caption = <figcaption>{this.caption}</figcaption>
      return (
        <figure>
          <img src={this.src} alt={this.alt} />
          {this.caption ? caption : ''}
        </figure>
      );
    }

    return '';
  }
}
