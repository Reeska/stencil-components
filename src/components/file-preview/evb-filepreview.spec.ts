import { EvbFilePreview } from './evb-file-preview';

describe('EvbFilePreview', () => {
  let component: EvbFilePreview;

  beforeEach(() => component = new EvbFilePreview());

  it('should render', () => {
    component.src = '/image.png';
    expect(component.render()).toMatchSnapshot();
  });

  it('should not render when image is not set', () => {
    component.src = undefined;
    expect(component.render()).toBe('');
    expect(component.render()).toMatchSnapshot();
  });
});
