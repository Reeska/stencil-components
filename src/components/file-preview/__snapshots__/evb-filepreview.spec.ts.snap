// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvbFilePreview should not render when image is not set 1`] = `""`;

exports[`EvbFilePreview should render 1`] = `
Object {
  "elm": undefined,
  "ishost": false,
  "vattrs": null,
  "vchildren": Array [
    Object {
      "elm": undefined,
      "ishost": false,
      "vattrs": Object {
        "alt": "preview",
        "src": "/image.png",
      },
      "vchildren": null,
      "vkey": undefined,
      "vname": undefined,
      "vtag": "img",
      "vtext": undefined,
    },
    Object {
      "vtext": "",
    },
  ],
  "vkey": undefined,
  "vname": undefined,
  "vtag": "figure",
  "vtext": undefined,
}
`;
